ambry-account/src/main/java/com/github/ambry/account/MySqlAccountService.java

```diff
@@ -0,0 +1,315 @@
+      return;
+    }
+
+    // get the last sync time of accounts and containers in cache
+    long lastModifiedTime = accountInfoMap.getLastModifiedTime();
+
+    try {
+      // Fetch all added/modified accounts and containers from MySql database since LMT
+      List<Account> accounts = mySqlAccountStore.getNewAccounts(lastModifiedTime);
+      List<Container> containers = mySqlAccountStore.getNewContainers(lastModifiedTime);
+      rwLock.writeLock().lock();
+      try {
+        accountInfoMap.updateAccounts(accounts);
+        accountInfoMap.updateContainers(containers);
+      } finally {
+        rwLock.writeLock().unlock();
+      }
+
+      // TODO: Find the max LMT in the fetched accounts and containers and update the cache
+
+    } catch (SQLException e) {
+      logger.error("Fetching Accounts from MySql DB failed", e);
+    }
+  }
+
+  @Override
+  public Account getAccountById(short accountId) {
+    rwLock.readLock().lock();
+    try {
+      return accountInfoMap.getAccountById(accountId);
+    } finally {
+      rwLock.readLock().unlock();
+    }
+  }
+
+  @Override
+  public Account getAccountByName(String accountName) {
+    rwLock.readLock().lock();
+    try {
+      return accountInfoMap.getAccountByName(accountName);
+    } finally {
+      rwLock.readLock().unlock();
+    }
+  }
+
+  @Override
+  public boolean updateAccounts(Collection<Account> accounts) {
+    Objects.requireNonNull(accounts, "accounts cannot be null");
+    if (accounts.isEmpty()) {
+      logger.debug("Empty account collection to update.");
+      return false;
+    }
+
+    if (mySqlAccountStore == null) {
+      logger.info("MySql Account DB store is not accessible");
+      return false;
+    }
+
+    if (config.updateDisabled) {
+      logger.info("Updates has been disabled");
+      return false;
+    }
+
+    if (hasDuplicateAccountIdOrName(accounts)) {
+      logger.error("Duplicate account id or name exist in the accounts to update");
+      //accountServiceMetrics.updateAccountErrorCount.inc();
+      return false;
+    }
+
+    // Make a pre check for conflict between the accounts to update and the accounts in the local cache. Will fail this
+    // update operation for all the accounts if any conflict exists. For existing accounts, there is a chance that the account to update
+    // conflicts with the accounts in the local cache, but does not conflict with those in the MySql database. This
+    // will happen if some accounts are updated but the local cache is not yet refreshed.
+    // TODO: Once we have APIs (and versioning) for updating containers, we will need to check conflicts for containers as well.
+    rwLock.readLock().lock();
+    try {
+      if (accountInfoMap.hasConflictingAccount(accounts)) {
+        logger.error("Accounts={} conflict with the accounts in local cache. Cancel the update operation.", accounts);
+        //accountServiceMetrics.updateAccountErrorCount.inc();
+        return false;
```

Comment:
I really wish the interface method threw an exception.  The caller has no way to tell if it failed due to bad input or system error.  We should revisit this later.